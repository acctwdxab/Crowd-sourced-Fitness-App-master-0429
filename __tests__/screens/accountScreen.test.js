import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import AccountScreen from '../../src/screens/accountScreen';

describe('<AccountScreen /> load', () => {
    it('has 1 child', () => {
        const tree = renderer.create(<AccountScreen />).toJSON();
        expect(tree.children.length).toBe(2);
    });

    it('renders correctly', async () => {
        const tree = renderer.create(<AccountScreen />).toJSON();
        await act(async () => { await expect(tree).toMatchSnapshot(); })
    });

});

describe('<AccountScreen /> functionality', () => {

    beforeEach(() => {
        user = {
            "-N-gWcl-XCle0sF_Zf7E": {
                email: "Test@email.com",
                firstName: "Joseph",
                lastName: "Deer",
                password: "testpass",
                profilePic: ""
            }
        }
        screen = render (
            <AccountScreen user={user} />
        );
    });

    it('First Name Edit Button', async () => {
        const { getByDisplayValue, getAllByText } = screen;
        const editButtons = await getAllByText('Edit');
        await fireEvent.press(editButtons[0]);
        const element = await getByDisplayValue('Joseph');
        expect(element).toBeTruthy();
    });

    it('Last Name Edit Button', async () => {
        const { getByDisplayValue, getAllByText } = screen;
        const editButtons = await getAllByText('Edit');
        await fireEvent.press(editButtons[1]);
        const element = await getByDisplayValue('Deer');
        expect(element).toBeTruthy();
    });

    it('Email Edit Button', async () => {
        const { getByDisplayValue, getAllByText } = screen;
        const editButtons = await getAllByText('Edit');
        await fireEvent.press(editButtons[2]);
        const element = await getByDisplayValue('Test@email.com');
        expect(element).toBeTruthy();
    });

    it('Password Edit Button', async () => {
        const { getByDisplayValue, getAllByText } = screen;
        const editButtons = await getAllByText('Edit');
        await fireEvent.press(editButtons[3]);
        const element = await getByDisplayValue('testpass');
        expect(element).toBeTruthy();
    });
});