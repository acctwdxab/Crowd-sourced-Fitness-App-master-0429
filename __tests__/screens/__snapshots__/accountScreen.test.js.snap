// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountScreen /> load renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "flex-start",
      "paddingTop": "20%",
    }
  }
>
  <View
    style={
      Object {
        "display": "flex",
        "justifyContent": "flex-start",
        "marginBottom": "10%",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 30,
        }
      }
    >
      Account
    </Text>
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignSelf": "center",
          "borderColor": "lightgrey",
          "borderRadius": 150,
          "borderWidth": 1,
          "height": 150,
          "justifyContent": "center",
          "marginBottom": "5%",
          "width": 150,
        }
      }
    >
      <View
        accessibilityLabel="Upload profile pic button"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(212, 0, 0)",
                },
              ]
            }
          >
            Upload Picture
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "lightgrey",
          "borderStyle": "solid",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "3%",
          "paddingLeft": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        no user logged in
      </Text>
      <View
        accessibilityLabel="Edit first name button"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(212, 0, 0)",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "lightgrey",
          "borderStyle": "solid",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "3%",
          "paddingLeft": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        no user logged in
      </Text>
      <View
        accessibilityLabel="Edit last name button"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(212, 0, 0)",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "lightgrey",
          "borderStyle": "solid",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "3%",
          "paddingLeft": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        no user logged in
      </Text>
      <View
        accessibilityLabel="Edit email button"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(212, 0, 0)",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderColor": "lightgrey",
          "borderStyle": "solid",
          "borderTopWidth": 1,
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": "3%",
          "paddingLeft": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 15,
          }
        }
      >
        *****************
      </Text>
      <View
        accessibilityLabel="Edit password button"
        accessibilityRole="button"
        accessibilityState={Object {}}
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
                Object {
                  "color": "rgb(212, 0, 0)",
                },
              ]
            }
          >
            Edit
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
